openssl/src/hash.rs

```diff
@@ -46,6 +47,25 @@ impl MessageDigest {
         }
     }
 
+    /// Returns the `MessageDigest` corresponding to an algorithm name.
+    ///
+    /// This corresponds to [`EVP_get_digestbyname`].
+    ///
+    /// [`EVP_get_digestbyname`]: https://www.openssl.org/docs/man1.1.0/crypto/EVP_DigestInit.html
+    pub fn from_name(name: &str) -> Option<MessageDigest> {
+        #[cfg(not(ossl110))]
```

Comment:
```suggestion
```

Init is already a no-op on versions that don't need it, so it can be called unconditionally.